utils/git.go

```diff
@@ -19,20 +22,26 @@ func GitClone(remoteURL string) (string, error) {
 		return "", err
 	}
 
-	clone := cloneArgs(remoteURL, root)
+	u, err := url.Parse(remoteURL)
+	if err != nil {
+		return "", err
+	}
 
-	if output, err := exec.Command("git", clone...).CombinedOutput(); err != nil {
+	fragment := u.Fragment
+	clone := cloneArgs(u, root)
+
+	if output, err := git(clone...); err != nil {
 		return "", fmt.Errorf("Error trying to use git: %s (%s)", err, output)
 	}
 
-	return root, nil
+	return checkoutGit(fragment, root)
 }
 
-func cloneArgs(remoteURL, root string) []string {
+func cloneArgs(remoteURL *url.URL, root string) []string {
 	args := []string{"clone", "--recursive"}
-	shallow := true
+	shallow := len(remoteURL.Fragment) == 0
```

Comment:
Why are we only doing a shallow clone when there's no fragment in the URL?  Is this a git restriction?